typespec-tests/tsp/response.tsp

```diff
@@ -92,4 +106,22 @@ interface ResponseOp {
     @header("operation-location")
     operationLocation: ResourceLocation<Resource>;
   };
+
+  @route("/exists")
+  @head
```

Comment:
Found the response model is boolean, will @head operation always return boolean response? is this handled by compiler or our emitter?

Reply:
No, not always. Batch got some other HEAD op that does not do this.

Reply:
who will resolve response model type, compiler?

Reply:
codegen. that part already exists https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/util/SchemaUtil.java#L99-L102

as well as azure-core

Reply:
got it.